flask_sse.py

```diff
@@ -135,10 +136,23 @@ def messages(self, channel='sse'):
         """
         pubsub = self.redis.pubsub()
         pubsub.subscribe(channel)
+        try:
+            for msg_dict in self.messages_loop(channel, pubsub):
+                yield Message(**msg_dict)
+        finally:
+            try:
+                pubsub.unsubscribe(channel)
+            except redis_exceptions.ConnectionError:
```

Comment:
Interesting... when would we have a `ConnectionError` here?

Reply:
If pubsub.listen() encounters a connection error e.g. if you stop the Redis server, the loop terminates due to an exception. We get "exception while handling exception" in pubsub.unsubscribe() in this scenario.